src/main/java/org/cryptomator/ui/mainwindow/ResizeController.java

```diff
@@ -86,6 +86,39 @@ private boolean didDisplayConfigurationChange() {
 		return configurationHasChanged;
 	}
 
+	private boolean isOutOfDisplayBounds() {
+		final int x = settings.windowXPosition.get();
+		final int y = settings.windowYPosition.get();
+		final int width = settings.windowWidth.get();
+		final int height = settings.windowHeight.get();
+
+		// Left pixel column
+		ObservableList<Screen> screens = Screen.getScreensForRectangle(x, y, 0, height);
+		if (screens.size() < 1) {
+			return true;
+		}
+
+		// Right pixel column
+		screens = Screen.getScreensForRectangle(x + width, y, 0, height);
+		if (screens.size() < 1) {
+			return true;
+		}
+
+		// Top pixel row
+		screens = Screen.getScreensForRectangle(x, y, width, 0);
+		if (screens.size() < 1) {
+			return true;
+		}
+
+		// Bottom pixel row
+		screens = Screen.getScreensForRectangle(x, y + height, width, 0);
+		if (screens.size() < 1) {
+			return true;
+		}
+
+		return false;
+	}
```

Comment:
Not sure if I've overoptimized this. 😅 And definitely haven't tested it. But maybe that's a more concise solution?

```suggestion
	private boolean isOutOfDisplayBounds() {
		final int x = settings.windowXPosition.get();
		final int y = settings.windowYPosition.get();
		final int width = settings.windowWidth.get();
		final int height = settings.windowHeight.get();

		return isRectangleOutOfScreen(x, y, 0, height) // Left pixel column
				|| isRectangleOutOfScreen(x + width, y, 0, height) // Right pixel column
				|| isRectangleOutOfScreen(x, y, width, 0) // Top pixel row
				|| isRectangleOutOfScreen(x, y + height, width, 0); // Bottom pixel row
	}

	private boolean isRectangleOutOfScreen(int x, int y, int width, int height) {
		return Screen.getScreensForRectangle(x, y, width, height).isEmpty();
	}
```

Reply:
Yes nice.